docs/docs/getting-started.server.mdx

```diff
@@ -573,40 +573,68 @@ on how to use MDX with Gatsby.
   <summary>Expand example</summary>
 
   ```js path="next.config.js"
-  module.exports = {
-    // Prefer loading of ES Modules over CommonJS
-    experimental: {esmExternals: true},
+  import nextMDX from '@next/mdx'
+
+  const withMDX = nextMDX({
+    // By default only the .mdx extension is supported.
+    extension: /\.mdx?$/,
+    options: {/* providerImportSource: …, otherOptions… */}
+  })
+
+  export default withMDX({
     // Support MDX files as pages:
     pageExtensions: ['md', 'mdx', 'tsx', 'ts', 'jsx', 'js'],
-    // Support loading `.md`, `.mdx`:
-    webpack(config, options) {
-      config.module.rules.push({
-        test: /\.mdx?$/,
-        use: [
-          // The default `babel-loader` used by Next:
-          options.defaultLoaders.babel,
-          {
-            loader: '@mdx-js/loader',
-            /** @type {import('@mdx-js/loader').Options} */
-            options: {/* jsxImportSource: …, otherOptions… */}
-          }
-        ]
-      })
-
-      return config
-    }
-  }
+  })
   ```
 </details>
 
-[Next](https://nextjs.org) supports webpack loaders by overwriting the webpack
-config in `next.config.js`.
+[Next.js](https://nextjs.org) have their own package for supporting MDX.
 
-Install and configure the webpack loader [`@mdx-js/loader`][webpack].
+Install and configure [`@next/mdx`][@next/mdx].
```

Comment:
I was under the impression that you still need `@mdx-js/loader` as well?

Reply:
It depends on your package manager. It’s a peer dependency. I [created an issue to propose properly migrating Next.js to MDX 2](https://github.com/vercel/next.js/issues/36888). This includes `@mdx-js/loader` to dependencies, but it was [moved to a discussion](https://github.com/vercel/next.js/discussions/36890) and has already moved to page 2, so I don’t think they want to prioritize this.

Since it’s a peer dependency, it’s installed implicitly when using newer npm versions, but not when using older npm versions or yarn.

Reply:
I believe newer npm versions require top-level users to install peer deps? 🤔 I believe in the past it didn’t, but now you do?

Reply:
No, it’s the other way around. Maybe you’re confused because MDX [.npmrc](https://github.com/mdx-js/mdx/blob/main/.npmrc) uses legacy peer deps?

Reply:
Weird. Not sure why I thought that.

Legacy peer deps are needed to show warnings but prevent errors. Many packages define incorrect peer ranges, and beta React version don’t work with peer ranges. npm 7 prevents them from being used.

Reply:
Should we mention on MDX 1 here though? I believe this is mostly a guide for users who are either configuring Next.js with MDX in a new project, or adding MDX support to an existing Next.js project. In both cases I think the focus should be MDX 2.

Reply:
nah, this is good!